### ISSUE TITLE
Frozen the statistics of BN layers?

### ISSUE THREAD

**Issue by nihaomiao** [speedinghzl/Pytorch-Deeplab#17]
Hello, Do you implement "frozen BN layers"? I find that the running means and running vars still keep changing during training.

**Comment by nihaomiao** [comment]
I have discussed with [isht7](https://github.com/isht7), and I find that the train.py in his code sets model to be evaluate mode firstly. But I find that your code sets the model to be train mode. This setting may keep changing the statistics of BN layers during training. I don't know whether I misunderstood your code.

**Comment by speedinghzl** [comment]
Yes，I keep changing the statistics of BN layers during training. This is very old code for semantic segmentation. Some details may be not maintained correctly. In fact, it's helpful to update the parameters and statistics of BN layers in the training stage with a large input size and batch size. Without sync BN, this code cannot achieve a higher performance. Maybe you could run the code with eval mode and train mode, and compare the performance.

Ps. I have implemented PSPnet, Deeplabv3, and Deeplabv3+ with high performance, I plan to release the code in November.

**Comment by nihaomiao** [comment]
OK, Thanks for your reply~

**Comment by sinemaslanx** [comment]
For small batch sizes, is it possible to frozen the statistics of BN layers? with which command is it possible?

**Comment by nihaomiao** [comment]
@siinem You can just set model.eval() to freeze the statistic. Please check the similar code from [isht7](https://github.com/isht7/pytorch-deeplab-resnet/blob/0c00ca1656dac8263b3d5db51910ee4fa0b10b49/train.py#L200).
